presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -137,6 +137,8 @@ public QueryObjectBundle rewriteQuery(@Language("SQL") String query, ClusterType
             Query createQuery = createTableAsSelect.getQuery();
             if (functionCallRewriter.isPresent()) {
                 createQuery = (Query) functionCallRewriter.get().rewrite(createQuery);
+                // Run zero row query to validate the rewritten query as early as REWRITE stage.
+                getResultMetadata(createQuery);
```

Comment:
Why is the validation done ? What happens if we don't do it ? Does getResultMetadata actually run say en explain on the rewritten query ?

Reply:
getResultMetadata() appends `limit 0` to the rewritten query and does run the rewritten query. Benefit of calling it QueryRewritter is that it can conclude the test case with the POSSIBLE_INVALID_FUNCTION_SUBSTITUTE error as early as the QueryStage.REWRITE, rather than running the actual query and failing at the QueryStage.CONTROL_MAIN.